src/docfx/lib/git/GitUtility.cs

```diff
@@ -35,6 +35,42 @@ public static (string url, string branch) GetGitRemoteInfo(string remoteHref)
             return (url, refSpec);
         }
 
+        /// <summary>
+        /// Execute git action with remote url with token and then reset the remote url back without token
+        /// </summary>
+        public static async Task WithToken(string cwd, string remoteUrl, string token, Func<string, Task> action)
+        {
+            if (string.IsNullOrEmpty(token))
+            {
+                await action(remoteUrl);
+                return;
+            }
+
+            // add token to remote href
+            var remoteUrlWithToken = AppendToken(remoteUrl, token);
+
+            await action(remoteUrlWithToken);
+
+            // reset url back without token
+            await SetRemoteGitUrl(cwd, remoteUrl);
+        }
+
+        /// <summary>
+        /// Append token to git remote href
+        /// </summary>
+        public static string AppendToken(string remoteHref, string token)
```

Comment:
>AppendToken [](start = 29, length = 11)

EmbedToken?

Reply:
good suggestion

---
In reply to: [202904142](https://github.com/dotnet/docfx/pull/3085#discussion_r202904142) [](ancestors = 202904142)